src/openforms/submissions/api/viewsets.py

```diff
@@ -304,7 +319,18 @@ def logic_check(self, request, *args, **kwargs):
             # keys like ``foo.bar`` and ``foo.baz`` are used which construct a foo object
             # with keys bar and baz.
             merged_data = {**submission_step.submission.data, **data}
-            evaluate_form_logic(submission_step, merged_data)
+            evaluate_form_logic(
+                submission_step.submission, submission_step, merged_data
+            )
 
-        serializer = self.get_serializer(instance=submission_step)
-        return Response(serializer.data)
+        submission_serializer = SubmissionSerializer(
+            instance=submission_step.submission, context={"request": request}
+        )
+        submission_step_serializer = self.get_serializer(instance=submission_step)
```

Comment:
`get_serializer_class` can be implement to look at `self.action` and return the appropriate serializer class based on that.

Reply:
This part of the code is no longer present because of the explicit serializer (see comment https://github.com/open-formulieren/open-forms/pull/602#discussion_r702941366)